spec/models/reservation_spec.rb

```diff
@@ -16,12 +16,17 @@
   it 'handles state change of reserved to shipped correctly' do
     @reservation.ship
     expect(@reservation.state).to eq('shipped')
+    #mailer tests here
+    last_email = ActionMailer::Base.deliveries.last
+    expect(last_email.subject).to include('You have purchased an item you reserved')
```

Comment:
this is a flaky spec waiting to happen, since shipping a gift sends a mailer to both the shipper and wediko - you should be testing for both of those emails here in any order